internal/mode/static/telemetry/collector.go

```diff
@@ -0,0 +1,78 @@
+package telemetry
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state"
+)
+
+type Collector interface {
+	CollectData() Data
+}
+
+type DataCollector struct {
+	ctx       context.Context
+	k8sClient client.Client
```

Comment:
similarly to the comment about ChangeProcessor, we don't need the whole client.Client. Could you  define an interface that only included the required methods?

This will be similar to  https://github.com/nginxinc/nginx-gateway-fabric/blob/9dd2968cb329ad3df94e6978f1e5f6a892375630/internal/framework/controller/getter.go#L13

Reply:
I added some changes which change it so we're using `client.Reader`, which makes sense to me. How does that sound to you? And is that doing what I'm hoping it to do -- with our `DataCollector` we will really only be reading from the Kubernetes API, and thus the `Get` and `List` methods on `client.Reader` is exactly what we need.

Reply:
client.Reader sounds good